src/util.rs

```diff
@@ -312,6 +313,90 @@ pub async fn unzip_file(
     Ok(ret)
 }
 
+pub async fn zip_files(
+    files: Vec<impl AsRef<Path>>,
+    dest: impl AsRef<Path>,
+    overwrite_old: bool,
+) -> Result<PathBuf, Error> {
+    let files = files.into_iter().map(|f| f.as_ref().to_path_buf()).collect::<Vec<PathBuf>>();
+    let mut dest = dest.as_ref().to_path_buf();
+
+    if !overwrite_old && dest.exists() {
+        dest = resolve_path_conflict(dest, None);
+    }
+
+    tokio::fs::create_dir_all(dest.parent().unwrap_or_else(|| Path::new("")))
+        .await
+        .context(format!("Failed to create directory {}", dest.display()))?;
```

Comment:
`unwrap_or_else(|| Path::new(""))` may not be correct. It's best to try to get the parent directory of the destination, and error if you can't